### PR TITLE
Changes for https://issues.openmrs.org/browse/FM-244

### PR DESCRIPTION
**Pull request by eunice18**
Add get active medication for a given patient uuid

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/fhir/api/MedicationRequestService.java [hunk 1]
```diff
@@ -35,6 +35,14 @@
...
     List<MedicationRequest> searchMedicationRequestById(String uuid);
 
+    /**
+     * Search drug orders list by uuid
+     *
+     * @param uuid of the drug order
+     * @return drug order obj
+     */
+    List<MedicationRequest> searchMedicationRequestByPatientId(String patientId);
+
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/module/fhir/api/MedicationRequestService.java]: parameter name and description doesn't match

File: api/src/main/java/org/openmrs/module/fhir/api/impl/MedicationRequestServiceImpl.java [hunk 2]
```diff
@@ -19,10 +19,7 @@
 import org.apache.commons.logging.LogFactory;
 import org.hl7.fhir.dstu3.model.IdType;
 import org.hl7.fhir.dstu3.model.MedicationRequest;
-import org.openmrs.CareSetting;
-import org.openmrs.Drug;
-import org.openmrs.DrugOrder;
-import org.openmrs.Order;
+import org.openmrs.*;
 import org.openmrs.api.context.Context;
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/module/fhir/api/impl/MedicationRequestServiceImpl.java]: do not use wild card imports

File: api/src/main/java/org/openmrs/module/fhir/api/util/FHIRMedicationRequestUtil.java [hunk 3]
```diff
@@ -147,8 +147,11 @@ public static MedicationRequest generateMedicationRequest(org.openmrs.DrugOrder
         if (doseUnit != null) {
             dose.setUnit(doseUnit.getName().getName());
         }
-        dose.setValue(omrsDrugOrder.getDose());
-        dosage.setDose(dose);
+
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/module/fhir/api/util/FHIRMedicationRequestUtil.java]: why this needs to be done?

File: omod/src/main/java/org/openmrs/module/fhir/providers/RestfulMedicationRequestProvider.java [hunk 4]
```diff
@@ -74,7 +75,18 @@ public MedicationRequest getResourceById(@IdParam IdType theId) {
 															@RequiredParam(name = MedicationRequest.SP_RES_ID) TokenParam id) {
 		return fhirMedicationRequestResource.searchByUniqueId(id);
 	}
-	
+
+	/**
+	 * Search Medication by unique id
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/providers/RestfulMedicationRequestProvider.java]: doc comment is wrong

File: api/src/main/java/org/openmrs/module/fhir/api/util/FHIRMedicationRequestUtil.java (hunk 3 above)
Comment (eunice18) [reply api/src/main/java/org/openmrs/module/fhir/api/util/FHIRMedicationRequestUtil.java]: this is because some drug orders contains dose in null

**Comment by eunice18** [comment]
Hello. I address your comments.

**Pull request merged**